File: 0874-backspace-string-compare/0874-backspace-string-compare.java
class Solution {
    public boolean backspaceCompare(String s, String t) {
        Stack<Character> a = new Stack();
        Stack<Character> b = new Stack();
        char[] sa = s.toCharArray();
        char[] ta = t.toCharArray();
        for(int i=0;i<s.length();i++){
            if(sa[i] == '#' && !a.isEmpty()){
                a.pop();
            }
            else if( sa[i]!= '#'){
                a.push(sa[i]);
            }
        }
        for(int i=0;i<t.length();i++){
            if(ta[i] == '#' && !b.isEmpty()){
                b.pop();
            }
            else if( ta[i]!= '#'){
                b.push(ta[i]);
            }
        }
        return a.equals(b);
    }
}

File: 0874-backspace-string-compare/0874-backspace-string-compare.py
class Solution(object):
    def backspaceCompare(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        stacka = []
        stackb = []

        for c in s:
            if stacka and c=='#':
                stacka.pop()
            elif c!='#':
                stacka.append(c)
        for c in t:
            if stackb and c=='#':
                stackb.pop()
            elif c!='#':
                stackb.append(c)
        return stacka==stackb
        